code/game/machinery/pipe/construction.dm

```diff
@@ -159,6 +163,16 @@ var/global/list/pipeID2State = list(
 
 	return
 
+/obj/item/pipe/AltClick(mob/user)
+	..()
+	if(!user.canUseTopic(user))

```

Comment:
I just copied it from here: https://github.com/tgstation/-tg-station/blob/b2948fcab7d2be5ce99210fcf5679cc9a886f48f/code/modules/clothing/head/soft_caps.dm#L21 (which is from my previous PR, so it might be wrong, but seems to work)